motivaa-backend/motivaa-app/src/test/java/backendtesting/testcases/ExcelOnModulesTest.java

```diff
@@ -1,74 +1,363 @@
+                new OutputVerification(excelOutput1Attribute, expectedValueAfterFirstUpdate),
+                new OutputVerification(excelOutput2Attribute,expectedValueAfterFirstUpdate)
+        );
+
+        TestUtils.updateProcessObjectAndVerifyOutputValues(
+                environmentHost,
+                processUpdateEndpoint,
+                jSessionId,
+                processObject,
+                excelInput1Attribute,
+                input1Value,
+                outputVerificationsForFirstUpdate
+        );
+
+        List<OutputVerification> outputVerificationsForSecondUpdate = Arrays.asList(
+                new OutputVerification(excelOutput1Attribute, input1Value),
+                new OutputVerification(excelOutput2Attribute,input2Value)
+        );
+
+        TestUtils.updateProcessObjectAndVerifyOutputValues(
+                environmentHost,
+                processUpdateEndpoint,
+                jSessionId,
+                processObject,
+                excelInput2Attribute,
+                input2Value,
+                outputVerificationsForSecondUpdate
+        );
+    }
+    @Test
+    void GIVEN_ExcelAttributeOnParent_InputsOnMultipleModules_WHEN_first_value_is_set_THEN_all_4_input_values_should_remain_empty() {
+        final String excelInput1Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule1Link.ExcelsOnModulesTestModule1Link::ExcelsOnModulesTestModule1.ExcelsOnModulesTestModule1Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule1Link::input1_shouldNotTriggerExcelUntilAll4InputsHaveValue_notOptional";
+        final String input1Value = "noValueShouldTriggerTheExcel";
+        final String excelOutput1Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule4Link.ExcelsOnModulesTestModule4Link::ExcelsOnModulesTestModule4.ExcelsOnModulesTestModule4Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule4Link::output1_shouldBeSameAsInput1";
+        final String excelOutput2Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule3Link.ExcelsOnModulesTestModule3Link::ExcelsOnModulesTestModule3.ExcelsOnModulesTestModule3Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule3Link::output2_shouldBeSameAsInput2";
+        final String excelOutput3Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule2Link.ExcelsOnModulesTestModule2Link::ExcelsOnModulesTestModule2.ExcelsOnModulesTestModule2Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule2Link::output3_shouldBeSameAsInput3";
+        final String excelOutput4Attribute = "ExcelsOnModulesTest.Modules.ExcelsOnModulesTestModule1Link.ExcelsOnModulesTestModule1Link::ExcelsOnModulesTestModule1.ExcelsOnModulesTestModule1Link::ExcelAttributeOnParent_InputsOnMultipleModules.ExcelsOnModulesTestModule1Link::output4_shouldBeSameAsInput4";
+        final String expectedValueAfterFirstUpdate = "";
```

Comment:
does the fact that it is the first update is an important information? If not maybe call it expectedValue, so it is easier to read it.